src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -971,6 +974,18 @@ public void EndBuild()
                             _overallBuildSuccess = false;
                         }
 
+                        if (loggingService is LoggingService logService)
```

Comment:
i assume in release the JIT does not bother emitting these conditions. ie., you don't need #if DEBUG.

Reply:
I am not confident, but I would guess so? I don't think it's a huge deal in this case if we're wrong, since the Debug.Asserts shouldn't fire.